lib/util/mutex.ts

```diff
@@ -19,17 +19,18 @@
  * limitations under the License.
  */
 
-'use strict';
+import { randomBytes } from 'crypto';
 
-const { randomBytes } = require('crypto');
+import Bluebird from 'bluebird';
 
-const Bluebird = require('bluebird');
-
-const kerror = require('../kerror');
-const debug = require('./debug')('kuzzle:mutex');
+import kerror from '../kerror';
+import buildDebug from './debug';
 
+const debug = buildDebug('kuzzle:mutex');
 const fatal = kerror.wrap('core', 'fatal');
 
+declare const kuzzle: any;
```

Comment:
`global` is not accessible directly with TS, we need to declare it to make it usable. And it seems that I didn't do it correctly.

Reply:
It does work, but only after the application has started. Otherwise, the kuzzle global is undefined.
What this line does is to make it know to the TS compiler that this global exists, it has no other meaning.

I've changed a bit how the kuzzle global is exposed: I've added a type file (`lib/util/Global.d.ts`) and asked the TS compiler to include it by default. So now, `global.kuzzle` is accessible from everywhere in our TS files, without having to do anything.

To prevent mistakes such as the one you made in your global review comment, I've also modified the `global.kuzzle` behavior a bit: now, if trying to access it before an application started, a meaningful error is thrown, telling the user that they have tried to access a live-only feature before starting their app.